File: src/main/java/net/teamhollow/direbats/client/model/DirebatsEntityModelLayers.java
package net.teamhollow.direbats.client.model;

import com.google.common.collect.ImmutableMap;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.fabricmc.fabric.api.client.rendering.v1.EntityModelLayerRegistry;
import net.minecraft.client.render.entity.model.EntityModelLayer;
import net.minecraft.util.Identifier;
import net.teamhollow.direbats.Direbats;
import net.teamhollow.direbats.client.model.entity.DirebatEntityModel;
import net.teamhollow.direbats.mixin.client.EntityModelLayersInvoker;

@Environment(EnvType.CLIENT)
public class DirebatsEntityModelLayers {
    public static final EntityModelLayer DIREBAT = registerMain("direbat");

    static {
        new ImmutableMap.Builder<EntityModelLayer, EntityModelLayerRegistry.TexturedModelDataProvider>()
            .put(DirebatsEntityModelLayers.DIREBAT, DirebatEntityModel::getTexturedModelData)
            .build().forEach(EntityModelLayerRegistry::registerModelLayer);
    }

    private static EntityModelLayer registerMain(String id) {
        return EntityModelLayersInvoker.invoke_register(new Identifier(Direbats.MOD_ID, id).toString(), "main");
    }
}

File: src/main/kotlin/dev/andante/direbats/client/render/entity/model/DirebatsEntityModelLayers.kt
package dev.andante.direbats.client.render.entity.model

import dev.andante.direbats.Direbats
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.fabricmc.fabric.api.client.rendering.v1.EntityModelLayerRegistry
import net.minecraft.client.render.entity.model.EntityModelLayer
import net.minecraft.util.Identifier

/**
 * Represents Direbats entity model layers.
 */
@Environment(EnvType.CLIENT)
object DirebatsEntityModelLayers {
    val DIREBAT = registerMain("direbat")

    private fun registerMain(id: String): EntityModelLayer {
        val layer = EntityModelLayer(Identifier(Direbats.MOD_ID, id), "main")
        EntityModelLayerRegistry.registerModelLayer(layer, DirebatEntityModel::TEXTURED_MODEL_DATA)
        return layer
    }
}
